process/smartContract/process.go

```diff
@@ -153,48 +187,93 @@ func (sc *scProcessor) ExecuteSmartContractTransaction(
 		return process.ErrNilSCDestAccount
 	}
 
-	err := sc.prepareSmartContractCall(tx, acntSnd)
+	err := sc.processSCPayment(tx, acntSnd)
 	if err != nil {
 		return err
 	}
 
+	defer func() {
+		if err != nil {
+			err = sc.processIfError(acntSnd, tx, vmcommon.UserError)
+			if err != nil {
+				log.Debug("error while processing error in smart contract processor")
+			}
+		}
+	}()
+
+	err = sc.prepareSmartContractCall(tx, acntSnd)
+	if err != nil {
+		return nil
+	}
+
 	vmInput, err := sc.createVMCallInput(tx)
 	if err != nil {
-		return err
+		return nil
 	}
 
 	vm, err := sc.getVMFromRecvAddress(tx)
 	if err != nil {
-		return err
+		return nil
 	}
 
 	vmOutput, err := vm.RunSmartContractCall(vmInput)
 	if err != nil {
-		return err
+		return nil
 	}
 
 	results, consumedFee, err := sc.processVMOutput(vmOutput, tx, acntSnd, round)
 	if err != nil {
-		return err
+		return nil
 	}
 
 	err = sc.scrForwarder.AddIntermediateTransactions(results)
 	if err != nil {
-		return err
+		return nil
 	}
 
 	sc.txFeeHandler.ProcessTransactionFee(consumedFee)
 
 	return nil
 }
 
-func (sc *scProcessor) prepareSmartContractCall(tx *transaction.Transaction, acntSnd state.AccountHandler) error {
-	err := sc.argsParser.ParseData(tx.Data)
+func (sc *scProcessor) processIfError(
+	acntSnd state.AccountHandler,
+	tx data.TransactionHandler,
+	returnCode vmcommon.ReturnCode,
+) error {
+	consumedFee := big.NewInt(0).SetUint64(tx.GetGasLimit() * tx.GetGasPrice())
```

Comment:
Do we really want to "eat" the whole txFee if an error happens?

Reply:
This is how bad processing in handled in eth, it is the most simple way. Otherwise, we have to treat this depending on errors and to add specific code inside Arwen.

Reply:
ok